### PR TITLE
docs(lsp): add examples to lsp-quickstart for completion and autoformatting

### PR DESCRIPTION
**Pull request by gpanders**
Auto-completion and auto-formatting are common (though certainly not universal) features that many users want. We can document how to accomplish this in lsp-quickstart so that users that do want these features can easily find examples of how to configure them.

### CODE REVIEW

**Review by MariaSolOs — APPROVED** [review]

**Review by mfussenegger — APPROVED** [review]
👍 

Should we also mention the snippet keymaps somehow?

**Comment by MariaSolOs** [comment]
> 👍
> 
> Should we also mention the snippet keymaps somehow?

Well those are set by default when a snippet session is active, so there's no need to do anything about them here.

**Comment by mfussenegger** [comment]
> > 👍
> > Should we also mention the snippet keymaps somehow?
> 
> Well those are set by default when a snippet session is active, so there's no need to do anything about them here.

But how are users supposed to know that that happens and which keys they can use?
Or that snippet expansion is even a thing

**Comment by MariaSolOs** [comment]
> But how are users supposed to know that that happens and which keys they can use? Or that snippet expansion is even a thing

Oh I see what you mean. I had misunderstood your original suggestion as mentioning "how to setup snippet keymaps".

**Comment by mfussenegger** [comment]
> > But how are users supposed to know that that happens and which keys they can use? Or that snippet expansion is even a thing
> 
> Oh I see what you mean. I had misunderstood your original suggestion as mentioning "how to setup snippet keymaps".

Sorry, I could've explained my point better.
I also just saw that the `vim.lsp.completion.enable` docs doesn't mention that aspect at all. It's not just the autotrigger, but opt-in to completion side effect handling - including snippet expansion. We should mention that somehow

**Comment by mfussenegger** [comment]
Merging this as it's already an improvement. We can follow up on the snippet-expansion and side effect aspects later

File: runtime/doc/lsp.txt [hunk 1]
```diff
@@ -34,16 +34,16 @@ Follow these steps to get LSP features:
     vim.api.nvim_create_autocmd('FileType', {
       -- This handler will fire when the buffer's 'filetype' is "python"
       pattern = 'python',
-      callback = function(ev)
+      callback = function(args)
```
Comment (justinmk) [inline runtime/doc/lsp.txt:37]: `ev` is an intentional convention to teach users that these args are event data.

**Pull request merged**